src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/CohostDocumentPullDiagnosticsEndpoint.cs

```diff
@@ -135,6 +139,24 @@ private async Task<VSInternalDiagnosticReport[]> HandleTaskListItemRequestAsync(
         ];
     }
 
+    private async Task<LspDiagnostic[]> GetCSharpTaskListItemsAsync(TextDocument razorDocument, CancellationToken cancellationToken)
+    {
+        if (!razorDocument.TryComputeHintNameFromRazorDocument(out var hintName))
+        {
+            return [];
+        }
+
+        var generatedDocument = await razorDocument.Project.TryGetSourceGeneratedDocumentFromHintNameAsync(hintName, cancellationToken).ConfigureAwait(false);
+        if (generatedDocument is null)
+        {
+            return [];
+        }
```

Comment:
Would be nice if this code didn't need to know about hintNames. Would it be possible to add an extension to TextDocument so we could just call something like:

var generatedDocument = await razorDocument.TryGetSourceGeneratedDocumentAsync(cancellationToken).ConfigureAwait(false);

Reply:
When I wrote diagnostics originally, I was on the fence about making this too easy, because accessing the source generated document in the devenv side of things (where this is) is generally undesirable. In fact, diagnostics and code actions are the only things that should, everything else should be on the remote side.

Having said that, an extension method that can't be called in OOP shouldn't be a problem, especially now that all of the code is written.

Reply:
Actually, its only diagnostics, so just a helper method in the base class would work here